### PR TITLE
Fix issue #171

### LINKED ISSUE #171: Intubation Checklist resets when traversing between tabs
**Steps to reproduce**
1. Go to Intubation Checklist 
2. Select cells under Teams
3. Go to any other tab and select cells, ie, Patient
4. Go back to Teams

Expected:
Selected cells should remain selected

Actual:
All cells have been deselected

checklist_bug

### PR DESCRIPTION
**Pull request by rockgecko-development**
Fix issue #171 by storing checked state in ChangeNotifierProvider class in IntubationChecklistPage

### CODE REVIEW

**Comment by workerbee22** [comment]
@rockgecko-development Looking good and this essentially sets our state management solution to Provider. Which I'm fine with. @maks you happy with Provider? BLOC seems way to heavy for this app.

**Comment by maks** [comment]
I actually use both Bloc & provider. I'm a big fan of provider, so 💯 ok with using it.

File: lib/widget/lists/intubation_checklist_item.dart [hunk 1]
```diff
@@ -22,7 +24,7 @@ class IntubationChecklistItemWidget extends StatefulWidget {
 
 class _IntubationChecklistItemWidgetState
     extends State<IntubationChecklistItemWidget> {
-  var _checked = false;
+  //var _checked = false;
```
Comment (maks) [inline lib/widget/lists/intubation_checklist_item.dart]: delete instead of commenting out?

File: lib/widget/lists/intubation_checklist_item.dart [hunk 2]
```diff
@@ -35,25 +37,27 @@ class _IntubationChecklistItemWidgetState
...
+        Provider.of<IntubationChecklistSelectionProvider>(context);
+    final isChecked = selectionState.isChecked(widget.listItem);
     return ReusableCardBase(
       elevation: 0,
       fallback: () {
-        setState(() {
-          _checked = !_checked;
-        });
+        // setState not needed, because Provider takes care of rebuilding.
+        selectionState.setChecked(widget.listItem, checked: !isChecked);
```
Comment (maks) [inline lib/widget/lists/intubation_checklist_item.dart]: can the item widget be stateless now too?

**Review by maks — CHANGES_REQUESTED** [review]
💯 👍 on switching to use provider for state @rockgecko-development

File: lib/widget/lists/intubation_checklist_item.dart (hunk 1 above)
Comment (rockgecko-development) [reply lib/widget/lists/intubation_checklist_item.dart]: ah forgot to clean that up

**Review by maks — APPROVED** [review]
:100: :+1:

**Pull request merged**